### DISCUSSION TITLE
Quick way to select partial word matches.

### DISCUSSION THREAD

**Discussion by feribg** [helix-editor/helix#7360] [Q&A] [answered]
Is there a way to toggle between full word and partial matches quickly without resorting to regex.

A common scenario is a set of variables that all contain a prefix `example_...` I want to do:
* select `example_` by select and `*`
* now enter visual mode and select all partial matches of example to refactor

Also the `*` and then `n` workflow of finding the selected word entering visual mode and selecting the next match seems to search for the next match from the begining of the file rather than the current line, is there a way to change that ?

This can be achieved with regex like \bexample_\b but is grossly inconvenient.

Thanks!

**Comment by the-mikedavis** [comment, marked as answer]
The `make_search_word_bounded` command should work for this. It isn't bound by default but you can call it from the command palette (`<space>?`) or bind it to a key in your config.

**Reply by feribg** [reply]
Thanks that works, any clue how to make the `next` in `search` mode continue from the current selection rather than from start of file ?

**Reply by the-mikedavis** [reply]
I see `search_next` (`n`) always searching after the current selection (unless it's the last and it wraps around). Do you see something different?
